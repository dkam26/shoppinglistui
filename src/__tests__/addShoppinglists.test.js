import React from 'react';
import { shallow, mount } from 'enzyme';
import {configure} from 'enzyme';
import Addshoppinglist from '../components/add.shoppinglist.component';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import toJson,{ shallowToJson } from 'enzyme-to-json';
import LocalStorageMock from '../setupTests';
import ReactDOM from 'react-dom';
configure({ adapter: new Adapter() });
describe('add shoopinglists component', () => {
    const props={
        getLists:()=>{},
        componentWillMount:()=>{},
        onSubmit :()=>{}
    }
    const wrapper =  shallow(<Addshoppinglist />);
    const preventDefault = jest.fn();
    it('should run', () => {
        expect(wrapper.exists(<div></div>)).toBe(true);
    });
    it('should render correctly', () => {
        const output = shallow( <Addshoppinglist />);
        expect(shallowToJson(output)).toMatchSnapshot();
        });
    it('should confirm states',()=>{
         expect(wrapper.state().shoppinglist).toBe('')
        //  expect(wrapper.state().).toBe('')
        
    
        })
    it('should add list',()=>{
            wrapper.find('#addShoppinglist').simulate('click')
            expect(toJson(wrapper)).toMatchSnapshot();
           
            
        });
    it('render the one input', () =>{
            expect(wrapper.find("#shoppingst")).toHaveLength(0);
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        });
    it('renders without crashing', () => {
            const div = document.createElement('div');
            ReactDOM.render(<Addshoppinglist />, div);
        }); 
    it('should ve the following methods',()=>{
            wrapper.instance().getLists();
            wrapper.instance().componentWillMount();
            wrapper.instance().onSubmit();
           });
})