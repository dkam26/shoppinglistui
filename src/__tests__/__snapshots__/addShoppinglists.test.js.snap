// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add shoopinglists component render the one input 1`] = `
<Redirect
  push={false}
  to="/shoppinglists"
/>
`;

exports[`add shoopinglists component should add list 1`] = `
<Redirect
  push={false}
  to="/shoppinglists"
/>
`;

exports[`add shoopinglists component should render correctly 1`] = `
<div>
  <Container>
    <Segment>
      <Menu
        secondary={true}
        style={
          Object {
            "backgroundColor": "black",
          }
        }
      >
        <MenuItem
          name="SHOPPINGLIST"
          style={
            Object {
              "color": "white",
            }
          }
        />
        <MenuMenu
          position="right"
        >
          <MenuItem
            name="logout"
            style={
              Object {
                "color": "white",
              }
            }
          />
        </MenuMenu>
      </Menu>
    </Segment>
    <Segment>
      <Menu
        secondary={true}
      >
        <MenuMenu>
          <MenuItem />
        </MenuMenu>
        <MenuMenu>
          <MenuItem>
            <h3
              style={
                Object {
                  "marginLeft": "308px",
                }
              }
            >
              Add new Shoppinglist
            </h3>
          </MenuItem>
        </MenuMenu>
        <MenuMenu
          position="right"
        >
          <MenuItem>
            <Button
              as="button"
              content="Back"
              icon="left chevron"
              id="addShoppinglist"
              labelPosition="left"
              onClick={[Function]}
            />
          </MenuItem>
        </MenuMenu>
      </Menu>
    </Segment>
    <div
      className="login-form"
    >
      <style>
        
      body &gt; div,
      body &gt; div &gt; div,
      body &gt; div &gt; div &gt; div.login-form {
        height: 100%;
      }
    
      </style>
      <Grid
        style={
          Object {
            "height": "100%",
          }
        }
        textAlign="center"
        verticalAlign="middle"
      >
        <GridColumn
          style={
            Object {
              "maxWidth": 450,
            }
          }
        >
          <Form
            as="form"
            size="large"
          >
            <Segment
              stacked={true}
            >
              <FormInput
                as={[Function]}
                control={[Function]}
                fluid={true}
                id="shoppingst"
                name="shoppinglist"
                onChange={[Function]}
                placeholder="shoppinglist"
              />
              <Button
                as="button"
                color="blue"
                fluid={true}
                onClick={[Function]}
                size="large"
              >
                Create shoppinglist
              </Button>
            </Segment>
          </Form>
        </GridColumn>
      </Grid>
    </div>
  </Container>
  <_class
    options={Object {}}
  />
</div>
`;
